### DISCUSSION TITLE
Why is my scss not used on the deployed website although it is used locally?

### DISCUSSION THREAD

**Discussion by durraniu** [quarto-dev/quarto-cli#11284] [Q&A] [answered]
### Description

I have my [quarto blog on github](https://github.com/durraniu/30_day_map_challenge_2024/tree/main). In the locally rendered blog, the hero image shows up and I can see it in the `_site` folder along with the scss. My `styles.scss` file specifies the hero image. But on the [rendered website on netlify](https://30daymapchallenge24.netlify.app/), I do not see the css applied. What am I doing wrong?

**Comment by mcanouil** [comment]
Because debugging a whole non default website is time consuming:

1. Upgrade to Quarto 1.6 release candidate, you'll see a lot of enhancement on how SCSS is handled on websites: https://quarto.org/docs/download/
2. Delete `.quarto`, `_site`, and `_freeze`.
3. render/publish again.

If this does not solve the issue, please try to come up with a simpler version, for example based on `quarto create project website`.

Note that the content of `_site` is exactly the same as the one you've published.
I can't see any differences.

| `_site` | Netlify |
|--------|--------|
| <img width="1624" alt="image" src="https://github.com/user-attachments/assets/c9f27165-30ae-4586-a2de-30ebfb4a264a"> | <img width="1624" alt="image" src="https://github.com/user-attachments/assets/d0a3c224-7aba-495e-a13b-e2edfd7c6046"> |
| <img width="1624" alt="image" src="https://github.com/user-attachments/assets/32ed5e52-a572-43d8-a26c-612e03876cec"> | <img width="1624" alt="image" src="https://github.com/user-attachments/assets/a39bfc97-7205-4613-9957-27d2138154b2"> |

**Reply by durraniu** [reply]
Thanks for looking into this. I updated quarto and deleted `.quarto`, `_site,` and `_freeze`. Re-rendering shows that the referenced image in scss is not copied to `_site`, although styles.scss is copied. I am not sure why.

**Reply by durraniu** [reply]
I am also not sure why `_freeze `is created as I'm not using the `freeze` option in the yaml file.

**Reply by mcanouil** [reply]
That is very confusing and misleading. You were talking about issue with your SCSS not being applied but now you are talking about a missing resource.

See Carlos' comment.

**Reply by durraniu** [reply]
Sorry, my intention was not to mislead. I followed the steps you suggested and shared the results of those steps. 
Please ignore the comment on `_freeze`.

**Reply by mcanouil** [reply]
I meant, your issue is the missing image that you defined in your SCSS, right?

**Reply by durraniu** [reply]
Yes, the image was not being copied.

**Reply by mcanouil** [reply]
Then the issue is exactly the same locally and on Netlify. As showed before, there are no differences.

Try adding explicitly the resource as showed in Carlos' comment.

**Comment by cscheid** [comment]
> My `styles.scss` file specifies the hero image

It's likely that something with our resource discovery code is not going as expected. A simple solution is to explicitly add the file to your resources list:

```yaml
website:
  resources:
    - filename1
    - filename2 # etc
```

**Reply by durraniu** [reply]
Thanks for this tip. I tried specifying the image file under `website` but `quarto preview` was throwing errors. So, I tried this:
```
project:
  type: website
  resources:
    - "image_title_block.jpg"
 ```
 It worked locally. The image is now copied to `_site`. I see this in RStudio viewer:
 
![image](https://github.com/user-attachments/assets/b33e4679-b040-4465-a24d-42d61d50c1a1)

But the image does not show up on netlify deployed site which is based on the `_site` on github. Not sure if it is quarto or netlify.

**Reply by mcanouil** [reply]
If the `_site` contains the image, and that the SCSS is correct.
Then you have to wait for your host provider to refresh the cache. You also have to refresh your cache. That's completely independent of Quarto.

**Reply by durraniu** [reply]
Netlify generally refreshes the site quickly after a push to github. But just to test if that was the problem, I did another deployment on netlify and got the same result. It seems that netlify somehow fails to recognize the scss file. I'll check scss file to see if something is wrong. Thanks for all the help.

**Reply by durraniu** [reply]
Quick update: Since I was not using any scss specific syntax, I renamed the file to styles.css and referenced that in the yml file. Now netlify recognized it and the image shows up.
My understanding was that quarto generates a css file from scss but I was wrong.

**Reply by mcanouil** [reply]
My website uses SCSS and is deployed on Netlify without any issues so unlikely to be the source of your issue.

**Reply by mcanouil** [reply, marked as answer]
Now, I understand `css` is for CSS not SCSS, so you can't put:
```
css: styles.scss
```

SCSS requires compilation, CSS does not.

See https://quarto.org/docs/output-formats/html-themes.html

Note that the image is present in  your `_site` so resource discovery seems to work.

**Reply by durraniu** [reply]
Thanks! And yes, the image was copied to _site after I explicitly specified that in the resources in yaml file. In the hindsight, the image may have been copied if I had correctly specified CSS/SCSS.
